qiskit/transpiler/passes/optimization/optimize_1q_gates.py

```diff
@@ -33,6 +33,15 @@
 class Optimize1qGates(TransformationPass):
     """Optimize chains of single-qubit u1, u2, u3 gates by combining them into a single gate."""
 
+    def __init__(self, basis=None):
+        """Optimize1qGates initializer.
+
+        Args:
+            basis (list[str]): Target basis names to unroll to, e.g. `['u3', 'cx']` .
```

Comment:
this can't be just any basis. this pass is very specific to working with 'u1' or 'u2' or 'u3'
```suggestion
            basis (list[str]): gates that this pass can emit. can be ['u1','u2','u3'] or ['u2','u3'] or ['u1','u3'] or ['u3']
```

Reply:
it can be any basis (see `test_optimize_u3_basis_u2_cx`). But it is true that "this pass is very specific to working with `'u1'`, `'u2'` and `'u3`". I tried to clarified here 5853c804f